db/db_basic_test.cc

```diff
@@ -849,23 +849,43 @@ TEST_F(DBBasicTest, MmapAndBufferOptions) {
 
 class TestEnv : public EnvWrapper {
   public:
-    explicit TestEnv(Env* base) : EnvWrapper(base) { };
+    explicit TestEnv() : EnvWrapper(Env::Default()),
+                close_count(0) { }
 
     class TestLogger : public Logger {
       public:
         using Logger::Logv;
+        TestLogger(TestEnv *env_ptr) : Logger() { env = env_ptr; }
+        ~TestLogger() {
+          if (!closed_) {
+            CloseHelper();
+          }
+        }
         virtual void Logv(const char *format, va_list ap) override { };
-      private:
+      protected:
         virtual Status CloseImpl() override {
+          return CloseHelper();
+        }
+      private:
+        Status CloseHelper() {
+          env->CloseCountInc();;
           return Status::NotSupported();
```

Comment:
I would think here status returned should be OK()

Reply:
Just wanted to return something unexpected in order to check that the return status is propagated to caller.